### PR TITLE
fix(backend): Remove Meilisearch index when notes are deleted

### LINKED ISSUE #11193: Indexed notes are not deleted in Meilisearch
## 💡 Summary

Indexed notes are not being deleted in Meilisearch, causing user anxiety and creating a burden on server management.

## 🥰 Expected Behavior

When a note is deleted, the corresponding indexed note in Meilisearch should also be deleted.

## 🤬 Actual Behavior

There is no functionality implemented in the server code to delete indexed notes.

## 📝 Steps to Reproduce

1. Activate Meilisearch on the server.
2. Create a note and then delete it.

### 🛰 Backend (for server admin)

* Misskey: 13.12.0 ~ 13.14.0-beta.2

### PR DESCRIPTION
**Pull request by caipira113**
## What
Resolve #11193
Added functionality to delete Meilisearch indexed notes when deleting a note.

## Why
When using Meilisearch on the server, indexed notes were not being removed even when the notes were deleted. This was not ideal for both privacy and server management.

## Additional info (optional)
At least in my local testing, it has been confirmed that the feature works on both local and remote.

## Checklist
- [x] Read the [contribution guide](https://github.com/misskey-dev/misskey/blob/develop/CONTRIBUTING.md)
- [x] Test working in a local environment
- [ ] (If needed) Add story of storybook
- [ ] (If needed) Update CHANGELOG.md
- [ ] (If possible) Add tests

### CODE REVIEW

File: packages/backend/src/core/NoteDeleteService.ts [hunk 1]
```diff
@@ -88,8 +91,8 @@ export class NoteDeleteService {
 			}
 
 			// also deliever delete activity to cascaded notes
-			const cascadingNotes = (await this.findCascadingNotes(note)).filter(note => !note.localOnly); // filter out local-only notes
-			for (const cascadingNote of cascadingNotes) {
+			const cascadingNotesFilter = (cascadingNotes).filter(note => !note.localOnly && note.userHost === null); // filter out local-only notes
```
Comment (saschanaz) [inline packages/backend/src/core/NoteDeleteService.ts]: federatedLocalCascadingNotes?

Comment (caipira113) [reply packages/backend/src/core/NoteDeleteService.ts]: Reflected in commit 0b6816b

File: packages/backend/src/core/NoteDeleteService.ts [hunk 2]
```diff
@@ -114,6 +117,11 @@ export class NoteDeleteService {
 			}
 		}
 
+		for (const cascadingNote of cascadingNotes) {
+			await this.searchService.unindexNote(cascadingNote);
```
Comment (syuilo) [inline packages/backend/src/core/NoteDeleteService.ts]: awaitする必要はないかも？
```suggestion
			this.searchService.unindexNote(cascadingNote);
```

**Comment by syuilo** [comment]
👍

**Comment by saschanaz** [comment]
Can this be tested? 🤔

**Pull request merged**